From ee24642f0e9be34e1db012efdb947966b448459d Mon Sep 17 00:00:00 2001
From: Don Zickus <dzickus@redhat.com>
Date: Tue, 16 Sep 2014 17:09:10 -0400
Subject: [usb] cdc-acm: remove redundant disconnected test from shutdown

Message-id: <1410887463-20674-479-git-send-email-dzickus@redhat.com>
Patchwork-id: 95465
O-Subject: [RHEL7 PATCH 478/591] USB: cdc-acm: remove redundant disconnected test from shutdown
Bugzilla: 1110939
RH-Acked-by: Tony Camuso <tcamuso@redhat.com>
RH-Acked-by: Prarit Bhargava <prarit@redhat.com>

https://bugzilla.redhat.com/show_bug.cgi?id=1110939

commit 89e54e4468338df5a4ab7627c5b8b10786ee43e8
Author: Johan Hovold <jhovold@gmail.com>
Date:   Mon May 26 19:23:47 2014 +0200

    USB: cdc-acm: remove redundant disconnected test from shutdown

    Remove redundant disconnect test from shutdown(), which is never called
    post disconnect() where we do synchronous hangup.

    Signed-off-by: Johan Hovold <jhovold@gmail.com>
    Signed-off-by: Greg Kroah-Hartman <gregkh@linuxfoundation.org>
---

Signed-off-by: Jarod Wilson <jarod@redhat.com>

diff --git a/drivers/usb/class/cdc-acm.c b/drivers/usb/class/cdc-acm.c
index bc7a2a6..91fdc29 100644
--- a/drivers/usb/class/cdc-acm.c
+++ b/drivers/usb/class/cdc-acm.c
@@ -594,30 +594,27 @@ static void acm_port_shutdown(struct tty_port *port)
 
  dev_dbg(&acm->control->dev, "%s\n", __func__);
 
- mutex_lock(&acm->mutex);
- if (!acm->disconnected) {
-  pm_err = usb_autopm_get_interface(acm->control);
-  acm_set_control(acm, acm->ctrlout = 0);
-
-  for (;;) {
-   urb = usb_get_from_anchor(&acm->delayed);
-   if (!urb)
-    break;
-   wb = urb->context;
-   wb->use = 0;
-   usb_autopm_put_interface_async(acm->control);
-  }
+ pm_err = usb_autopm_get_interface(acm->control);
+ acm_set_control(acm, acm->ctrlout = 0);
 
-  usb_kill_urb(acm->ctrlurb);
-  for (i = 0; i < ACM_NW; i++)
-   usb_kill_urb(acm->wb[i].urb);
-  for (i = 0; i < acm->rx_buflimit; i++)
-   usb_kill_urb(acm->read_urbs[i]);
-  acm->control->needs_remote_wakeup = 0;
-  if (!pm_err)
-   usb_autopm_put_interface(acm->control);
+ for (;;) {
+  urb = usb_get_from_anchor(&acm->delayed);
+  if (!urb)
+   break;
+  wb = urb->context;
+  wb->use = 0;
+  usb_autopm_put_interface_async(acm->control);
  }
- mutex_unlock(&acm->mutex);
+
+ usb_kill_urb(acm->ctrlurb);
+ for (i = 0; i < ACM_NW; i++)
+  usb_kill_urb(acm->wb[i].urb);
+ for (i = 0; i < acm->rx_buflimit; i++)
+  usb_kill_urb(acm->read_urbs[i]);
+
+ acm->control->needs_remote_wakeup = 0;
+ if (!pm_err)
+  usb_autopm_put_interface(acm->control);
 }
 
 static void acm_tty_cleanup(struct tty_struct *tty)
-- 
1.7.1